File: frontend/src/services/authService.ts
import { api } from '@/lib/api'
import type { User } from '@/types'

export interface LoginPayload { email: string; password: string }
export interface RegisterPayload {
  first_name: string
  last_name: string
  middle_name?: string
  email: string
  phone: string
  password: string
  password_confirmation: string
  municipality_id: number
  barangay_id: number
}

export const authService = {
  login: (payload: LoginPayload) =>
    api.post<{ message: string; token: string; user: User }>('/auth/login', payload),

  register: (payload: RegisterPayload) =>
    api.post<{ message: string; user: User; verification_code?: string; resumed?: boolean }>('/auth/register', payload),

  verifyOtp: (email: string, otp: string) =>
    api.post<{ message: string; token: string; user: User }>('/auth/verify-otp', { email, otp }),

  resendOtp: (email: string) =>
    api.post<{ message: string; verification_code?: string }>('/auth/resend-otp', { email }),

  forgotPassword: (email: string) => api.post<{ message: string }>('/auth/forgot-password', { email }),

  resetPassword: (payload: { email: string; token: string; password: string; password_confirmation: string }) =>
    api.post<{ message: string }>('/auth/reset-password', payload),

  me: () => api.get<{ user: User }>('/auth/me'),

  changePassword: (payload: { current_password: string; password: string; password_confirmation: string }) =>
    api.post<{ message: string }>('/auth/change-password', payload),

  logout: () => api.post<{ message: string }>('/auth/logout'),

  logoutAllDevices: () => api.post<{ message: string }>('/auth/logout-all'),
}

File: mobile/lib/services/auth_service.dart
import '../core/api/api_client.dart';
import '../models/user.dart';

class AuthService {
  AuthService(this._api);

  final ApiClient _api;

  Future<({String token, AppUser user, String message})> login({
    required String email,
    required String password,
  }) {
    return _api.handle(
      _api.post('/auth/login', data: {
        'email': email,
        'password': password,
        'device_name': 'AgriConnect Mobile',
      }),
      (json) {
        final map = json as Map<String, dynamic>;
        return (
          token: map['token'] as String,
          user: AppUser.fromJson(map['user'] as Map<String, dynamic>),
          message: map['message'] as String? ?? 'Login successful.',
        );
      },
    );
  }

  Future<AppUser> me() {
    return _api.handle(
      _api.get('/auth/me'),
      (json) => AppUser.fromJson((json as Map<String, dynamic>)['user'] as Map<String, dynamic>),
    );
  }

  Future<void> logout() async {
    try {
      await _api.post('/auth/logout');
    } catch (_) {
      // Clear local session even if network fails.
    }
  }

  Future<void> changePassword({
    required String currentPassword,
    required String password,
    required String passwordConfirmation,
  }) {
    return _api.handle(
      _api.post('/auth/change-password', data: {
        'current_password': currentPassword,
        'password': password,
        'password_confirmation': passwordConfirmation,
      }),
      (_) {},
    );
  }
}
